client.c

```diff
@@ -922,13 +922,20 @@ int fio_clients_send_ini(const char *filename)
 int fio_client_update_options(struct fio_client *client,
 			      struct thread_options *o, uint64_t *tag)
 {
-	struct cmd_add_job_pdu pdu;
+	size_t cmd_sz = offsetof(struct cmd_add_job_pdu, top) +
+		thread_options_pack_size(o);
+	struct cmd_add_job_pdu *pdu;
+	int ret;
 
-	pdu.thread_number = cpu_to_le32(client->thread_number);
-	pdu.groupid = cpu_to_le32(client->groupid);
-	convert_thread_options_to_net(&pdu.top, o);
+	pdu = malloc(cmd_sz);
```

Comment:
Should we check for malloc() failure?

Reply:
The vast majority of fio code assumes that malloc() always succeeds. 

On 64bit Linux, this is mostly a safe assumption (unless the user is running with restrictive rlimits).

While I'd normally prefer to check for malloc failure, it's better to follow the style of the code. Plus some of the calls I'm adding have no way to bubble up an error.